### PR TITLE
Vision test

### PR DESCRIPTION
**Pull request by JosephTLockwood**
Add Limelight Vision and Sim to the Main. It also adds PathPlanner Path and Choreo Path.

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/drive/Drive.java [hunk 1]
```diff
@@ -54,7 +56,12 @@ public class Drive extends SubsystemBase {
   private final Module[] modules = new Module[4]; // FL, FR, BL, BR
 
   private SwerveDriveKinematics kinematics = new SwerveDriveKinematics(getModuleTranslations());
+
+  @Getter
+  @Setter
```
Comment (CrispyBacon1999) [inline src/main/java/frc/robot/subsystems/drive/Drive.java:61]: This blows my mind that it exists lol

File: src/main/java/frc/robot/subsystems/drive/GyroIONavX2.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  public void updateInputs(GyroIOInputs inputs) {
+    inputs.connected = navx.isConnected();
+    inputs.yawPosition = Rotation2d.fromDegrees(navx.getYaw());
+    inputs.yawVelocityRadPerSec = Units.degreesToRadians(navx.getRawGyroZ());
+    inputs.odometryYawPositions =
+        yawPositionQueue.stream().map(Rotation2d::fromDegrees).toArray(Rotation2d[]::new);
+
+    yawPositionQueue.clear();
+  }
+}
```
Comment (CrispyBacon1999) [inline src/main/java/frc/robot/subsystems/drive/GyroIONavX2.java:30]: Is this just for completion's sake, making it easier for other people?

File: readme.md [hunk 3]
```diff
@@ -1,28 +1,43 @@
...
-
-
-## Credits
-
-This project is heavily based on example code from Team 6328 - Mechanical Advantage from their [AdvantageKit repository](https://github.com/Mechanical-Advantage/AdvantageKit)
+![Hemlock's Gray Matter - AdvantageKit Swerve Base](./docs/images/Github%20Header.png)
+
+<p align="center">
+  <a href="https://github.com/wpilibsuite/allwpilib/releases/tag/v2024.1.1-beta-4"><img src="https://img.shields.io/badge/WPILib-2024.1.1--beta--4-AC2B37" /></a>
+  <a href="https://v6.docs.ctr-electronics.com/en/stable/"><img src="https://img.shields.io/badge/Phoenix6-24.0.0--beta--5-97D700"></a>
```
Comment (CrispyBacon1999) [inline readme.md:5]:
```suggestion
  <a href="https://v6.docs.ctr-electronics.com/en/stable/"><img src="https://img.shields.io/badge/Phoenix6-24.0.0--beta--7-97D700"></a>
```

File: src/main/java/frc/robot/Constants.java [hunk 4]
```diff
@@ -24,7 +24,7 @@
 public final class Constants {
   public static final Mode currentMode = Mode.SIM;
 
-  public static enum Mode {
+  public enum Mode {
```
[the remark this answers is not in the document]
Comment (JosephTLockwood) [reply src/main/java/frc/robot/Constants.java:27]: Yeah, enums are static. I downloaded Sonarlint and it was recommended to change.

File: src/main/java/frc/robot/subsystems/drive/Drive.java (hunk 1 above)
Comment (JosephTLockwood) [reply src/main/java/frc/robot/subsystems/drive/Drive.java:61]: Not sure if I like it or not yet but think I'll get used to it.

File: src/main/java/frc/robot/subsystems/drive/GyroIONavX2.java (hunk 2 above)
Comment (JosephTLockwood) [reply src/main/java/frc/robot/subsystems/drive/GyroIONavX2.java:30]: Make it easier for other people. Steven's team is interested in using it, so I figured I might as well take a shot.

File: readme.md (hunk 3 above)
Comment (JosephTLockwood) [reply readme.md:5]: Some of this stuff maybe better moved to a wiki on how to setup and use.

**Pull request merged**